### ISSUE TITLE
GAN is not working

### ISSUE THREAD

**Issue by RobertoSanz88** [Francesco601/machine-learning-stuff#1]
I get the error.
You must compile your model before training/testing. Use `model.compile(optimizer, loss)`.

If I add a couple of lines like this:
generator.compile(optimizer=keras.optimizers.Adam(learning_rate=0.0002), loss='binary_crossentropy')
discriminator.compile(optimizer=keras.optimizers.Adam(learning_rate=0.0002), loss='binary_crossentropy')

It starts running (or doing something) but the final 10 images are just NOISE. 
Is that the desired result? I hoped to see a number generated....

Roberto.

**Comment by Francesco601** [comment]
Which GAN? What is the name of the actual file?

**Comment by Francesco601** [comment]
Ok, it's been a while but I  made some corrections and it should be fine,

**Comment by RobertoSanz88** [comment]
Hi again,

I run your code in Colab and this is what I get, in attach. Is this supposed to be OK? Seems noise to me.
![Imagen1](https://github.com/Francesco601/machine-learning-stuff/assets/85118625/dfc0553d-71f9-4f05-8b74-a7404adda6ea)

**Comment by Francesco601** [comment]
Yeah, that is what I'm getting. I don't know  what the problem is.

**Comment by Francesco601** [comment]
I'm working on a different approach using PyTorch. I will upload the file when I have some time. Thanks for bringing this to my attention in any case.

**Comment by Francesco601** [comment]
Ok, it works now but your images are saved to a directory called runs.
